client/src/reducers/CartReducer.js

```diff
@@ -10,30 +10,23 @@ const CartReducer = () => {
     const { isUserloggedIn,uid } = useAuth();
 
     useEffect( () => {
+      if(!isUserloggedIn){
+        return;
+      }
       ( async function(){
         try {
           const res = await axios.get("https://shopping-hub-2021.herokuapp.com/api/carts/"+uid);
```

Comment:
Use template literals here,it's the better approach

Reply:
yeah trying to figure it out working on making a template literal for async calls